### PR TITLE
Support `current_geometry_index` in route refresh

### PR DESCRIPTION
**Pull request by chezzdev**

### CODE REVIEW

File: Sources/MapboxDirections/Directions.swift [hunk 1]
```diff
@@ -439,8 +439,28 @@ open class Directions: NSObject {
...
      */
     @discardableResult open func refreshRoute(responseIdentifier: String, routeIndex: Int, fromLegAtIndex startLegIndex: Int = 0, completionHandler: @escaping RouteRefreshCompletionHandler) -> URLSessionDataTask? {
+        _refreshRoute(responseIdentifier: responseIdentifier,
+                      routeIndex: routeIndex,
+                      fromLegAtIndex: startLegIndex,
+                      currentRouteShapeIndex: nil,
+                      completionHandler: completionHandler)
+    }
+
+    @discardableResult open func refreshRoute(responseIdentifier: String, routeIndex: Int, fromLegAtIndex startLegIndex: Int = 0, currentRouteShapeIndex: Int, completionHandler: @escaping RouteRefreshCompletionHandler) -> URLSessionDataTask? {
```
Comment (1ec5) [inline Sources/MapboxDirections/Directions.swift:449]: I assume we’re overloading the method instead of giving `currentRouteShapeIndex` a default value to avoid changing the method signature. We should document the new method and explain the difference between the two.

**Review by 1ec5 — COMMENTED** [review]
Remember to add a changelog entry for the new publicly documented methods and also add tests to ensure we don’t introduce off-by-one errors.

**Review by 1ec5 — APPROVED** [review]
